### PR TITLE
Upload status window that displays information about the file and upl…

### PR DESCRIPTION
**Pull request by qwangd2l**
…oad progress.

This "window" resides in the bottom right of the app's screen.
Here are some screenshots:

Some in progress and some are done:
image
Failed and finished:
image
Lesser items means lower height (there's a max height for the container):
image
Minimized:
image
What it looks like in the whole layout of the page:
image
Scrolling (position fixed):
image

### CODE REVIEW

File: src/components/upload-status-management.js [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+		} else if (this.uploader.uploads.length > 1) {
+			message = this.localize('uploadsCompleted', 'uploadsCompleted', this.uploader.uploads.filter(upload => upload.progress === 100).length);
+		} else {
+			message = '';
+		}
+
+		return html`<span role="status" aria-live="polite" class="header-title d2l-heading-4">${message}</span>`;
+	}
+
+	renderUploadProgress(upload) {
```
Comment (AdmiralCoco) [inline src/components/upload-status-management.js]: Nit: can destructure upload (`renderUploadProgress({ progress, file })`)

**Comment by rcausey** [comment]
Looks great! What happens when you upload additional objects? Do they get added to the list?

What happens when you close the popup and then upload additional objects? Is it a fresh list?

**Comment by rcausey** [comment]
Can you make the "Upload Failed" text closer to the title, so there's equal space below the "Upload Failed" text and above the title?

**Comment by qwangd2l** [comment]
> Looks great! What happens when you upload additional objects? Do they get added to the list?
> 
> What happens when you close the popup and then upload additional objects? Is it a fresh list?

Yeah, additional objects get added tot he list.
When the popup is closed and you upload additional objects:
- In addition to the new objects appearing in the list, objects that are in progress from before will remain. 
- Failed or finished uploads will be removed from the list when reopening

**Comment by rcausey** [comment]
> > Looks great! What happens when you upload additional objects? Do they get added to the list?
> > What happens when you close the popup and then upload additional objects? Is it a fresh list?
> 
> Yeah, additional objects get added tot he list.
> When the popup is closed and you upload additional objects:
> 
> * In addition to the new objects appearing in the list, objects that are in progress from before will remain.
> * Failed or finished uploads will be removed from the list when reopening

Perfect

**Comment by rcausey** [comment]
> Can you make the "Upload Failed" text closer to the title, so there's equal space below the "Upload Failed" text and above the title?

Also can you make the "Upload Failed" text smaller? "Small Body Text" style? http://design.d2l/style/typography/

**Comment by qwangd2l** [comment]
> > Can you make the "Upload Failed" text closer to the title, so there's equal space below the "Upload Failed" text and above the title?
> 
> Also can you make the "Upload Failed" text smaller? "Small Body Text" style? http://design.d2l/style/typography/

It's actually already using small body text.

> Can you make the "Upload Failed" text closer to the title, so there's equal space below the "Upload Failed" text and above the title?

Yes, I'll try doing that

**Pull request merged**